src/OrchardCore.Modules/OrchardCore.Contents/Liquid/ContentAnchorTag.cs

```diff
@@ -140,53 +136,41 @@ public async ValueTask<Completion> WriteToAsync(List<FilterArgument> argumentsLi
             }
             else if (adminFor != null)
             {
-                contentItem = adminFor;
-                var metadata = await contentManager.PopulateAspectAsync<ContentItemMetadata>(adminFor);
+                var metadata = await PopulateAspectForContentItemMetadataAsync(adminFor);
 
-                if (metadata.AdminRouteValues != null)
+                if (metadata.AdminRouteValues != null && routeValues != null)
                 {
-                    if (routeValues != null)
+                    foreach (var attribute in routeValues)
                     {
-                        foreach (var attribute in routeValues)
-                        {
-                            metadata.AdminRouteValues.Add(attribute.Key, attribute.Value);
-                        }
+                        metadata.AdminRouteValues.Add(attribute.Key, attribute.Value);
                     }
 
                     customAttributes["href"] = urlHelper.Action(metadata.AdminRouteValues["action"].ToString(), metadata.AdminRouteValues);
                 }
             }
             else if (removeFor != null)
             {
-                contentItem = removeFor;
-                var metadata = await contentManager.PopulateAspectAsync<ContentItemMetadata>(removeFor);
+                var metadata = await PopulateAspectForContentItemMetadataAsync(removeFor);
 
-                if (metadata.RemoveRouteValues != null)
+                if (metadata.RemoveRouteValues != null && routeValues != null)
                 {
-                    if (routeValues != null)
+                    foreach (var attribute in routeValues)
                     {
-                        foreach (var attribute in routeValues)
-                        {
-                            metadata.RemoveRouteValues.Add(attribute.Key, attribute.Value);
-                        }
+                        metadata.RemoveRouteValues.Add(attribute.Key, attribute.Value);
                     }
 
                     customAttributes["href"] = urlHelper.Action(metadata.RemoveRouteValues["action"].ToString(), metadata.RemoveRouteValues);
                 }
             }
             else if (createFor != null)
             {
-                contentItem = createFor;
-                var metadata = await contentManager.PopulateAspectAsync<ContentItemMetadata>(createFor);
+                var metadata = await PopulateAspectForContentItemMetadataAsync(createFor);
```

Comment:
Here you are passing createFor just like before.

Reply:
Sure